pkg/tsdb/opentsdb/opentsdb_test.go

```diff
@@ -1,16 +1,80 @@
 package opentsdb
 
 import (
+	"io/ioutil"
+	"net/http"
+	"strings"
 	"testing"
 
 	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 func TestOpenTsdbExecutor(t *testing.T) {
 	exec := &OpenTsdbExecutor{}
 
+	t.Run("create request", func(t *testing.T) {
+		req, err := exec.createRequest(&models.DataSource{}, OpenTsdbQuery{})
+		require.NoError(t, err)
+
+		assert.Equal(t, "POST", req.Method)
+		body, err := ioutil.ReadAll(req.Body)
+		require.NoError(t, err)
+
+		testBody := "{\"start\":0,\"end\":0,\"queries\":null}"
+		assert.Equal(t, testBody, string(body))
+	})
+
+	t.Run("Parse response should handle invalid JSON", func(t *testing.T) {
+		response := `{ invalid }`
+
+		query := OpenTsdbQuery{}
+
+		result, err := exec.parseResponse(query, &http.Response{Body: ioutil.NopCloser(strings.NewReader(response))})
+		require.Nil(t, result["A"].Dataframes)
+		require.Error(t, err)
+	})
+
+	t.Run("Parse response should handle JSON", func(t *testing.T) {
+		response := `
+		[
+			{
+				"metric": "test",
+				"dps": {
+					"58": 50.0
+				}
+			}
+		]`
+
+		query := OpenTsdbQuery{}
+
+		resp := http.Response{Body: ioutil.NopCloser(strings.NewReader(response))}
+		resp.StatusCode = 200
+		result, err := exec.parseResponse(query, &resp)
+		require.NoError(t, err)
+
+		decoded, err := result["A"].Dataframes.Decoded()
+		require.NoError(t, err)
+		require.Len(t, decoded, 1)
+
+		frame := decoded[0]
+
+		require.Len(t, frame.Fields, 2)
+
+		field1 := frame.Fields[0]
```

Comment:
it is not mandatory, and not used for most of the cases yet. but might be nice if we could use go-cmp instead. going to do the same when sdk changes for SQLs